ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/PullPhysicalPlan.java

```diff
@@ -138,8 +137,7 @@ private boolean requiresRequestsToAllPartitions() {
     return lookupConstraints.stream()
         .anyMatch(lookupConstraint -> {
           if (lookupConstraint instanceof KeyConstraint) {
-            final KeyConstraint keyConstraint = (KeyConstraint) lookupConstraint;
-            return keyConstraint.getConstraintOperator() != ConstraintOperator.EQUAL;
```

Comment:
I didn't follow the reason for this change. It seems like range scans still do require requests to all partitions, right?

Reply:
Good catch, thx!